src/Components/animation/ProgressBar.js

```diff
@@ -11,226 +16,235 @@ class ProgressBar extends React.Component {
+                    left: item.left,
+                  },
+                ]}
+                key={item.key + this.state.stateFromProps.progressBarDots.length}
+              />
+            ))}
 
-    // Width of the progress bar
-    this.progressBarWidth = this.props.animationWidth - 120;
-    var stepWidth = this.progressBarWidth / (this.props.stepCount - 1);
+        {/* Play button */}
+        <TouchableOpacity
+          style={[styles.playBtn, { left: this.state.stateFromProps.playLeft }]}
+          onPress={() => this.props.playAnimation()}
+        >
+          <Image style={styles.playIcn} source={iconPlay} />
+        </TouchableOpacity>
+
+        {/* Black bar */}
+        <Animated.View
+          style={[
+            styles.black,
+            styles.progressBar,
+            this.props.readonly && styles.largeProgressBar,
+            { width: this.state.stateFromProps.interpolatedWidth },
+          ]}
+        />
 
-    // If the bar is in the editor, there is an additional button to add a step
-    if (!this.props.readonly) {
-      stepWidth = this.progressBarWidth / this.props.stepCount;
-      this.progressBarWidth -= stepWidth;
-    }
+        {/* Clickable areas to move to each step */}
+        {this.state.stateFromProps.progressBarDots?.map((item, idx, arr) => (
+          <TouchableOpacity
+            style={
+              this.props.readonly
+                ? [
+                    styles.largeHitBox,
+                    {
+                      width: item.touchableWidth,
+                      left: item.touchableLeft,
+                    },
+                  ]
+                : [styles.dotHitBox, { left: item.left - DOT_SIZE / 2 }]
+            }
+            onPress={() => this.props.goToStep(item.key)}
+            key={1000 + item.key}
+          />
+        ))}
+
+        {/* White squares */}
+        {!this.props.readonly
+          ? undefined
+          : this.state.stateFromProps.progressBarDots?.map(item => (
+              <Animated.View
+                style={[styles.white, styles.square]}
+                key={item.key + this.state.stateFromProps.progressBarDots.length}
+                left={item.left + 3}
+              />
+            ))}
+
+        {/* Buttons: add step, remove step, trash */}
+        {!this.props.readonly ? (
+          <View style={[StyleSheet.absoluteFill]} height="100%" width="100%">
+            <TouchableOpacity style={[styles.dot, { right: 60 }]} onPress={() => this.props.onStepAdded()}>
+              <Image style={styles.controlIcn} source={iconAdd} />
+            </TouchableOpacity>
+            <TouchableOpacity style={[styles.dot, { right: 35 }]} onPress={() => this.props.onStepRemoved()}>
+              <Image style={styles.controlIcn} source={iconMinus} />
+            </TouchableOpacity>
+            <TouchableOpacity style={[styles.dot, { right: 10 }]}>
+              <Image style={styles.controlIcn} source={iconTrash} />
```

Comment:
I think we could use icons directly, so that we don't have to maintain images in our assets folder for this.

If needed we can do it together

Reply:
Do you have an example where an icon is used instead of an image?